addon/components/listing.js

```diff
@@ -146,6 +146,118 @@ export default class ListingComponent extends Component {
     this.renderSubForms();
   }
 
+  @action
+  moveUp(subform) {
+    const orderedSubForms = this.orderedSubForms;
+    const index = this.orderedSubForms.indexOf(subform);
+
+    if (index === 0) {
+      return;
+    }
+
+    const previousSubform = orderedSubForms.at(index - 1);
+
+    const currentOrder = getOrder(
+      subform.sourceNode,
+      this.formStore,
+      this.graphs.sourceGraph
+    );
+    const newOrder = getOrder(
+      previousSubform.sourceNode,
+      this.formStore,
+      this.graphs.sourceGraph
+    );
+
+    this.formStore.removeStatements([
+      {
+        subject: subform.sourceNode,
+        predicate: ORDER,
+        object: currentOrder,
+        graph: this.graphs.sourceGraph,
+      },
+      {
+        subject: previousSubform.sourceNode,
+        predicate: ORDER,
+        object: newOrder,
+        graph: this.graphs.sourceGraph,
+      },
+    ]);
+
+    this.formStore.addAll([
+      {
+        subject: subform.sourceNode,
+        predicate: ORDER,
+        object: newOrder,
+        graph: this.graphs.sourceGraph,
+      },
+      {
+        subject: previousSubform.sourceNode,
+        predicate: ORDER,
+        object: currentOrder,
+        graph: this.graphs.sourceGraph,
+      },
+    ]);
+
+    // eslint-disable-next-line no-self-assign
+    this.subForms = this.subForms; // Force a rerender with the new ordering
+  }
+
+  @action
+  moveDown(subform) {
+    const orderedSubForms = this.orderedSubForms;
+    const index = this.orderedSubForms.indexOf(subform);
+
+    if (index === orderedSubForms.length) {
+      return;
+    }
+
+    const nextSubform = orderedSubForms.at(index + 1);
+
+    const currentOrder = getOrder(
+      subform.sourceNode,
+      this.formStore,
+      this.graphs.sourceGraph
+    );
+    const newOrder = getOrder(
+      nextSubform.sourceNode,
+      this.formStore,
+      this.graphs.sourceGraph
+    );
+
+    this.formStore.removeStatements([
+      {
+        subject: subform.sourceNode,
+        predicate: ORDER,
+        object: currentOrder,
+        graph: this.graphs.sourceGraph,
+      },
+      {
+        subject: nextSubform.sourceNode,
+        predicate: ORDER,
+        object: newOrder,
+        graph: this.graphs.sourceGraph,
+      },
+    ]);
+
+    this.formStore.addAll([
+      {
+        subject: subform.sourceNode,
+        predicate: ORDER,
+        object: newOrder,
+        graph: this.graphs.sourceGraph,
+      },
+      {
+        subject: nextSubform.sourceNode,
+        predicate: ORDER,
+        object: currentOrder,
+        graph: this.graphs.sourceGraph,
+      },
+    ]);
```

Comment:
This will trigger the "observers" twice, once for each operation. Ideally that only happens once but that requires changes in the forking-store library.

Reply:
I added a TODO comment.